solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/FlowReceiverContainer.java

```diff
@@ -67,28 +70,42 @@ public UUID bind() throws JCSMPException {
 		// Ideally would just use flowReceiverReference.compareAndSet(null, newFlowReceiver),
 		// but we can't initialize a FlowReceiver object without actually provisioning it.
 		// Explicitly using a lock here is our only option.
-		Lock writeLock = readWriteLock.writeLock();
 		writeLock.lock();
 		try {
-			logger.info(String.format("Binding flow receiver container %s", id));
+			LOGGER.info("Binding flow receiver container {}", id);
 			FlowReceiverReference existingFlowReceiverReference = flowReceiverAtomicReference.get();
 			if (existingFlowReceiverReference != null) {
 				UUID existingFlowRefId = existingFlowReceiverReference.getId();
-				logger.info(String.format("Flow receiver container %s is already bound to %s", id, existingFlowRefId));
+				LOGGER.info("Flow receiver container {} is already bound to {}", id, existingFlowRefId);
 				return existingFlowRefId;
 			} else {
-				logger.info(String.format("Flow receiver container %s started in state '%s'", id, isPaused.get() ? "Paused" : "Running"));
+				LOGGER.info("Flow receiver container {} started in state '{}'", id, isPaused.get() ? "Paused" : "Running");
 				consumerFlowProperties.setStartState(!isPaused.get());
-				consumerFlowProperties.addRequiredSettlementOutcomes(Outcome.ACCEPTED, Outcome.FAILED, Outcome.REJECTED);
 
-				FlowReceiver flowReceiver = session.createFlow(null, consumerFlowProperties, endpointProperties, eventHandler);
-				if (eventHandler != null && eventHandler instanceof SolaceFlowHealthEventHandler) {
-					((SolaceFlowHealthEventHandler) eventHandler).setHealthStatusUp();
+				if (!transacted) {
+					consumerFlowProperties.addRequiredSettlementOutcomes(Outcome.ACCEPTED, Outcome.FAILED, Outcome.REJECTED);
+				}
+
+				TransactedSession transactedSession = transacted ? session.createTransactedSession() : null;
+
+				try {
+					FlowReceiver flowReceiver = transactedSession != null ?
+							transactedSession.createFlow(null, consumerFlowProperties, endpointProperties, eventHandler) :
+							session.createFlow(null, consumerFlowProperties, endpointProperties, eventHandler);
+					if (eventHandler != null && eventHandler instanceof SolaceFlowHealthEventHandler) {
```

Comment:
is creation of a flow sufficient to assume that flow is up?  should it not be started first?

Reply:
> is creation of a flow sufficient to assume that flow is up?

Yes.

>  should it not be started first?

No, because flows may be created in a paused state, which can occur if a binding is `paused` -> `stopped` -> `started`. See: https://github.com/SolaceProducts/solace-spring-cloud/blob/486c5d3cba82f7441b57bfd4a3e057369dfca3db/solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/FlowReceiverContainer.java#L82-L83